### PR TITLE
Disjoint Sets: Quick Find implementation

### PR DESCRIPTION
**Pull request by antonioaversa**

### CODE REVIEW

File: MoreStructures/DisjointSets/QuickFindDisjointSet.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    /// <inheritdoc/>
+    /// <remarks>
+    /// Calculated in constant-time by comparing the set id of the first value with the set id of the second value.
+    /// <br/>
+    /// Time and Space Complexity are O(1).
+    /// </remarks>
+    public bool AreConnected(int first, int second)
+    {
+        if (first < 0 || first >= ValuesCount)
+            throw new ArgumentException(
```
Comment (andreaguarino) [inline MoreStructures/DisjointSets/QuickFindDisjointSet.cs:52]: _Nitpick_ : I'd suggest having a dedicated error message when the set is empty

File: MoreStructures.Tests/DisjointSets/QuickFindDisjointSetTests.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        Assert.AreEqual(4, disjointSet.SetCount);
+    }
+
+    [TestMethod]
+    public void Ctor_RaisesExceptionOnInvalidValuesCount()
+    {
+        Assert.ThrowsException<ArgumentException>(() => Builder(-1));
+        Assert.ThrowsException<ArgumentException>(() => Builder(-100));
+    }
+}
```
Comment (andreaguarino) [inline MoreStructures.Tests/DisjointSets/QuickFindDisjointSetTests.cs:33]: Add a new line

File: MoreStructures/DisjointSets/QuickFindDisjointSet.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        var secondSetId = Find(second);
+
+        if (firstSetId == secondSetId)
+            return;
+
+        for (var i = 0; i < ValuesCount; i++)
+            if (SetIds[i] == secondSetId)
+                SetIds[i] = firstSetId;
+    }
+}
```
Comment (andreaguarino) [inline MoreStructures/DisjointSets/QuickFindDisjointSet.cs:94]: Add a new line

**Review by andreaguarino — APPROVED** [review]

**Comment by antonioaversa** [comment]
All comments addressed

**Review by andreaguarino — APPROVED** [review]

**Pull request merged**